notebooks/237-typo-detector/237-typo-detector.ipynb

```diff
@@ -0,0 +1,537 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tokenizer = AutoTokenizer.from_pretrained(model_id)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Then we use the inference pipeline for `token-classification` task. You can find more information about usage Hugging Face inference pipelines in this [tutorial](https://huggingface.co/docs/transformers/pipeline_tutorial)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "nlp = pipeline('token-classification', model=model, tokenizer=tokenizer, aggregation_strategy=\"average\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Function to find typos in a sentence and write them to the terminal"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def show_typos(sentence):\n",
+    "\n",
+    "    typos = [sentence[r[\"start\"]: r[\"end\"]] for r in nlp(sentence)]\n",
+    "\n",
+    "    detected = sentence\n",
+    "    for typo in typos:\n",
+    "        detected = detected.replace(typo, f'<i>{typo}</i>')\n",
+    "\n",
+    "    print(\"[Input]: \", sentence)\n",
+    "    print(\"[Detected]: \", detected)\n",
+    "    print(\"-\" * 130)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Demo"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sentences = [\n",
+    "    \"He had also stgruggled with addiction during his time in Congress .\",\n",
+    "    \"The review thoroughla assessed all aspects of JLENS SuR and CPG esign maturit and confidence .\",\n",
+    "    \"Letterma also apologized two his staff for the satyation .\",\n",
+    "    \"Vincent Jay had earlier won France 's first gold in gthe 10km biathlon sprint .\",\n",
+    "    \"It is left to the directors to figure out hpw to bring the stry across to tye audience .\",\n",
+    "]\n",
+    "\n",
+    "for sentence in sentences:\n",
+    "    show_typos(sentence)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 2. Converting the model to ONNX and then to OpenVino IR"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### Load the Pytorch model\n",
+    "\n",
+    "Use the `AutoModelForTokenClassification` class to load the pretrained pytorch model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model_id = \"m3hrdadfi/typo-detector-distilbert-en\"\n",
+    "\n",
+    "tokenizer = AutoTokenizer.from_pretrained(model_id)\n",
+    "model_dir = Path(\"model2\")\n",
```

Comment:
Maybe name the model folder with more meaningful names? It seems more clear why we need to name two separate folders for storing models.